roles/splunk_common/tasks/add_forward_server.yml

```diff
@@ -17,24 +17,21 @@
   no_log: "{{ hide_password }}"
 
 - name: "Enable ssl-forwarding to {{ forward_servers }}"
-  uri:
-    url: "{{ cert_prefix }}://127.0.0.1:{{ splunk.svc_port }}/servicesNS/nobody/system/configs/conf-outputs"
-    method: POST
-    user: "{{ splunk.admin_user }}"
-    password: "{{ splunk.password }}"
-    validate_certs: false
-    body:
-      name: "tcpout:group1"
-      server: "{% for fwd in forward_servers %}{{ fwd }}:{{ splunk.s2s.port }}{{ ',' if not loop.last else '' }}{% endfor %}"
-      clientCert: "{{ splunk.s2s.cert if splunk.s2s is defined and splunk.s2s.cert is defined else '' }}"
-      sslPassword: "{{ splunk.s2s.password if splunk.s2s is defined and splunk.s2s.password is defined else '' }}"
-      useClientSSLCompression: "{{ 'true' if splunk.s2s is defined and splunk.s2s.ssl is defined and splunk.s2s.ssl else '' }}"
-    body_format: "form-urlencoded"
-    status_code: 201,409
-    timeout: 10
-  when: '"s2s" in splunk and "ssl" in splunk.s2s and splunk.s2s.ssl | bool'
-  register: forward_status
-  changed_when: forward_status.status == 201
+  become: yes
+  become_user: "{{ splunk.user }}"
+  ini_file:
+    path: "{{ splunk.home }}/etc/system/local/outputs.conf"
+    section: "tcpout:group1"
+    option: "{{ item.key }}"
+    value: "{{ item.value }}"
+  with_items:
+    - {key: "server", value: "{% for fwd in forward_servers %}{{ fwd }}:{{ splunk.s2s.port }}{{ ',' if not loop.last else '' }}{% endfor %}"}
+    - {key: "clientCert", value: "{{ splunk.s2s.cert if splunk.s2s is defined and splunk.s2s.cert is defined else ''}}"}
+    - {key: "sslPassword", value: "{{ splunk.s2s.password if splunk.s2s is defined and splunk.s2s.password is defined else ''}}"}
+    - {key: "useClientSSLCompression", value: "{{ 'true' if splunk.s2s is defined and splunk.s2s.ssl is defined and splunk.s2s.ssl else '' }}"}
+  when:
+    - '"s2s" in splunk and "ssl" in splunk.s2s and splunk.s2s.ssl | bool'
+    - item.value | length > 0
   notify:
```

Comment:
Just a note here on idempotency: I don't believe we can use `notify` to trigger Splunk to restart. I believe what happens here is when this play runs for the first time and creates the outputs.conf file, the status will be "changed" and the "restart Splunk" handler will get triggered. But at the time of execution, Splunk isn't running - shortly after we call the Splunk start playbooks/tasks. But the notify handler doesn't get cleared, so at the end of the site.yml Splunk will restart when it doesn't need to.

At the same time, we should leave in some mechanism for triggering restarts too because some people may be using these playbooks outside of the context of docker-splunk. You may be calling `site.yml` on an already-running Splunk instance, in which case you will want to trigger the handler.

I haven't found a good way of handling this other than what I did here https://github.com/splunk/splunk-ansible/blob/develop/roles/splunk_common/tasks/s2s/configure_splunktcp.yml

Reply:
Okay, pushing a change to include the trigger you have instead